### ISSUE TITLE
Function spacemacs/rustic-cargo-repeat missing in rust layer

### ISSUE THREAD

**Issue by emlun** [syl20bnr/spacemacs#16203]
#### Description :octocat:

Function `spacemacs/rustic-cargo-repeat` missing in rust layer

Commit 3879dc3e55a78274c3cac61758742e96052fb130 overhauls the rust layer, migrating to Rustic as the new Rust mode. The updated code defines the key mapping `LEADER c .` for `spacemacs/rustic-cargo-repeat`, but this function does not seem to exist anywhere in Git history. Instead, the `LEADER c .` command results in the message:

```
command-execute: Wrong type argument: commandp, spacemacs/rustic-cargo-repeat
```

Indeed, `describe-key` for this key mapping prints:

```
help-fns--analyze-function: Symbol’s function definition is void: spacemacs/rustic-cargo-repeat
```

#### Reproduction guide :beetle:
- Start Emacs
- Open scratch buffer
- Enable `rustic-mode`
- Enter the key command `LEADER c .` (for example `, c .` in my config)

*Observed behaviour:* :eyes: :broken_heart:
```
command-execute: Wrong type argument: commandp, spacemacs/rustic-cargo-repeat
```

*Expected behaviour:* :heart: :smile:
The last `rustic-cargo-` command ran executes again.

#### System Info :computer:
- OS: gnu/linux
- Emacs: 29.1
- Spacemacs: 0.999.0
- Spacemacs branch: develop (rev. cb3b93711)
- Graphic display: t
- Running in daemon: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
(systemd protobuf go lua cucumber
         (google-java-format :variables google-java-format/format-on-save t)
         octave shell-scripts csv sql docker rust nginx ruby html markdown javascript
         (typescript :variables typescript-indent-level 2)
         (java :variables java-backend 'nope)
         yaml python clojure vimscript colors
         (scala :variables scala-backend 'scala-metals)
         asciidoc react emoji auto-completion emacs-lisp git helm lsp multiple-cursors syntax-checking treemacs
         (version-control :variables version-control-diff-side 'left version-control-global-margin t)
         terraform)
```
- System configuration features: ACL CAIRO DBUS FREETYPE GIF GLIB GMP GNUTLS GPM GSETTINGS HARFBUZZ JPEG JSON LCMS2 LIBOTF LIBSYSTEMD LIBXML2 M17N_FLT MODULES NOTIFY INOTIFY PDUMPER PNG RSVG SECCOMP SOUND SQLITE3 THREADS TIFF TOOLKIT_SCROLL_BARS TREE_SITTER WEBP X11 XDBE XIM XINPUT2 XPM GTK3 ZLIB

**Comment by smile13241324** [comment]
I have found another non existing function, which was referencing to the layers `funcs.el` instead of to the rustic package. However I do not find an alternative for `rustic-cargo-repeat`, which I have deactivated for now.

@whirm can you take a look and add the function back if it is available?

**Comment by smile13241324** [comment]
Maybe it would be worth a shot if you two @whirm @emlun could help review https://github.com/syl20bnr/spacemacs/pull/16217 which tries to build upon the changes on the rust layer as I am not a rust user.

**Comment by smile13241324** [comment]
I have merged #16217 which should solve this issue.
